crates/chain/tests/test_tx_graph_conflicts.rs

```diff
@@ -0,0 +1,1282 @@
+#[macro_use]
+mod common;
+
+use std::collections::{BTreeMap, BTreeSet, HashSet};
+
+use bdk_chain::{
+    indexed_tx_graph::{self, IndexedTxGraph},
+    keychain::{self, Balance, KeychainTxOutIndex},
+    local_chain::LocalChain,
+    tx_graph, BlockId, ChainPosition, ConfirmationHeightAnchor, SpkIterator,
+};
+use bitcoin::{
+    hashes::Hash, secp256k1::Secp256k1, BlockHash, OutPoint, Script, ScriptBuf, Transaction, TxIn,
+    TxOut,
+};
+use common::*;
+use miniscript::{Descriptor, DescriptorPublicKey};
+
+#[allow(dead_code)]
+struct Scenario<'a> {
+    /// Name of the test scenario
+    name: &'a str,
+    /// Transaction templates
+    tx_templates: &'a [TxTemplate<'a, ConfirmationHeightAnchor>],
```

Comment:
Since #1069 is merged now, we should use `BlockId` as the anchor to simplify these tests.